test_harness/src/hebench_ibenchmark.cpp

```diff
@@ -319,7 +323,7 @@ void PartialBenchmarkDescriptor::describe(const Engine &engine,
 
     description.workload_name = completed_description.workload_name;
     description.header        = ss.str();
-    description.path          = ss_path;
+    description.path          = std::string(ss_path);
```

Comment:
I know the UtilityPath class was written to handle and overload a bunch of stuff here, but I'd suggest keep original code and simply replace the path separators by dashes in the resulting string when the single path is requested. This may cause less changes and make maintenance easier by using the standard C++ as much as possible.